src/server/page_server.cpp

```diff
@@ -557,17 +567,17 @@ page_server::disconnect_tran_server_async (const connection_handler *conn)
 	      m_async_disconnect_handler.disconnect (std::move (*it));
 	      assert (*it == nullptr);
 	      m_passive_tran_server_conn.erase (it);
-	      passive_tran_server_found = true;
 	      break;
 	    }
 	}
-      assert (passive_tran_server_found);
     }
 }
 
 void
 page_server::disconnect_all_tran_server ()
 {
+  std::lock_guard lk_guard (m_conn_mutex);
```

Comment:
I think the scope is too large. Especially, the dtor for a connection_handler is quite a long job, which even waits for joining internal threads. However, I just wrap them all now because they are going to be changed in the following issue ([LETS-639](http://jira.cubrid.org/browse/LETS-639))

Reply:
I didn't know that a new connection can be established even after `finalize_server_type()`. If it's the case, we should block it. It may cause the new connection to suffer from communication errors right after establishing the connection. I think it should be dealt with in another issue.

Reply:
`finalize_server_type` server type is called after `disconnect_all_tran_server` (see `xboot_shutdown_server`).
There is a small window of opportunity. In practice it may never be a problem.
But, as you say, a PTS booting up in this window of opportunity will most probably have to deal with it.
Lets forget about this for now.
If you want, you can add a new issue about this: a shutting down `page_server` must block new connections to be made.
Up to you.

Reply:
Ok, let's forget about this for now.